packages/python-sdk/e2b/sandbox_async/sandbox_api.py

```diff
@@ -66,32 +89,89 @@ async def list(
                 metadata = urllib.parse.urlencode(quoted_metadata)
 
         async with AsyncApiClient(config) as api_client:
-            res = await get_sandboxes.asyncio_detailed(
+            res = await get_v2_sandboxes.asyncio_detailed(
                 client=api_client,
                 metadata=metadata,
+                state=query.state or UNSET,
+                limit=limit,
+                next_token=next_token,
             )
 
-        if res.status_code >= 300:
-            raise handle_api_exception(res)
+            if res.status_code >= 300:
+                raise handle_api_exception(res)
 
-        if res.parsed is None:
-            return []
+            if res.parsed is None:
+                return ListSandboxesResponse(
+                    sandboxes=[],
+                    has_more_items=False,
+                    next_token=None,
+                    iterator=cls._list_iterator(query=query, api_key=api_key, domain=domain, debug=debug, request_timeout=request_timeout)
+                )
 
-        return [
-            SandboxInfo(
-                sandbox_id=SandboxApi._get_sandbox_id(
-                    sandbox.sandbox_id,
-                    sandbox.client_id,
-                ),
-                template_id=sandbox.template_id,
-                name=sandbox.alias if isinstance(sandbox.alias, str) else None,
-                metadata=(
-                    sandbox.metadata if isinstance(sandbox.metadata, dict) else {}
-                ),
-                started_at=sandbox.started_at,
+            token = res.headers.get("x-next-token")
+            has_more_items = bool(token)
+            sandboxes = [
+                SandboxInfo(
+                    sandbox_id=SandboxApi._get_sandbox_id(
+                        sandbox.sandbox_id,
+                        sandbox.client_id,
+                    ),
+                    template_id=sandbox.template_id,
+                    name=sandbox.alias if isinstance(sandbox.alias, str) else None,
+                    metadata=(
+                        sandbox.metadata if isinstance(sandbox.metadata, dict) else {}
+                    ),
+                    started_at=sandbox.started_at,
+                    state=sandbox.state,
+                )
```

Comment:
this could be some util function, you are doing the same in sync Python

Reply:
it is a class shared by both